logsmanagement/db_api.c

```diff
@@ -0,0 +1,1398 @@
+
+#define LOGS_MANAG_DB_VERSION 1
+
+static sqlite3 *main_db = NULL;   /**< SQLite DB handler for MAIN_DB **/
+static char *main_db_dir = NULL;  /**< Directory where all the log management databases and log blobs are stored in **/
+static char *main_db_path = NULL; /**< Path of MAIN_DB **/
+
+/* -------------------------------------------------------------------------- */
+/*                            Database migrations                             */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * @brief No-op database migration, just to bump up starting version.
+ * @param database Unused
+ * @param name Unused
+ * @return Always 0.
+ */
+static int do_migration_noop(sqlite3 *database, const char *name){
+    UNUSED(database);
+    UNUSED(name);
+    collector_info("Running database migration %s", name);
+    return 0;
+}
+
+typedef struct database_func_migration_list{
+    char *name;
+    int (*func)(sqlite3 *database, const char *name);
+} DATABASE_FUNC_MIGRATION_LIST;
+
+DATABASE_FUNC_MIGRATION_LIST migration_list_main_db[] = {
+    {.name = MAIN_DB" v0 to v1",  .func = do_migration_noop},
+    // the terminator of this array
+    {.name = NULL, .func = NULL}
+};
+
+DATABASE_FUNC_MIGRATION_LIST migration_list_metadata_db[] = {
+    {.name = METADATA_DB_FILENAME " v0 to v1",  .func = do_migration_noop},
+    // the terminator of this array
+    {.name = NULL, .func = NULL}
+};
+
+typedef enum {
+    ERR_TYPE_OTHER,
+    ERR_TYPE_SQLITE,
+    ERR_TYPE_LIBUV,
+} logs_manag_db_error_t;
+
+/**
+ * @brief Logs a database error
+ * @param[in] log_source Log source that caused the error
+ * @param[in] error_type Type of error
+ * @param[in] rc Error code
+ * @param[in] line Line number where the error occurred (__LINE__)
+ * @param[in] file Source file where the error occurred (__FILE__)
+ * @param[in] func Function where the error occurred    (__FUNCTION__)
+ */
+static void throw_error(const char *const log_source, 
+                        const logs_manag_db_error_t error_type,
+                        const int rc, const int line, 
+                        const char *const file, const char *const func){
+    collector_error("[%s]: %s database error: (%d) %s (%s:%s:%d))",
+                        log_source ? log_source : "-", 
+                        error_type == ERR_TYPE_OTHER ? "" : ERR_TYPE_SQLITE ? "SQLite" : "libuv",
+                        rc, error_type == ERR_TYPE_OTHER ? "" : ERR_TYPE_SQLITE ? sqlite3_errstr(rc) : uv_strerror(rc), 
+                        file, func, line);
+}
+
+/**
+ * @brief Get version of SQLite
+ * @return String that contains the SQLite version, or NULL in case of error.
+ *         Must be freed.
+ */
+char *db_get_sqlite_version() {
+    int rc = 0;
+    sqlite3_stmt *stmt_get_sqlite_version;
+    rc = sqlite3_prepare_v2(main_db, "SELECT sqlite_version();", -1, &stmt_get_sqlite_version, NULL);
```

Comment:
I was reviewing this function, and  I was thinking about the final result. @Dim-P, why are you not using the [library](https://www.sqlite.org/c3ref/libversion.html) to get the final result when `netdata` is compiled?

Reply:
Good point, I will remove the function completely as it's used only for debugging and use the macro instead.